### PR TITLE
SAK-30087 & SAK-29824 - Clean up redundant attributes from submission.

### PR DESCRIPTION
**Pull request by jonespm**
Would also fix the bug with this being invalid and not maching in
assignment xml

### CODE REVIEW

**Comment by jonespm** [comment]
I tested this locally on most of the pages, the student submission, instructor grading. It looks okay. I don't see anywhere these were really needed. I think that having all of those inner classes in BaseAssignmentService (like BaseAssignmentSubmission) also lead to possible problems with redundant code in AssignmentSubmissionAccess (which is fixed here) but I figured we could maybe refactor that after so this merge could be possible.

**Comment by jonespm** [comment]
Based on some feedback rewrote this using this new method addElementsToList that further simplifies this code. I believe the AssignmentSubmissionAccess will also be collapsed eventually to use this service method but that inner class still needs to be pulled out.

File: assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentSubmissionAccess.java [hunk 1]
```diff
@@ -128,6 +129,47 @@ public AssignmentSubmissionAccess()
 		
 	}
 
+	/**
+	 * Helper method to add elements or attributse to a list
+	 * @param attributeName Name of the attribute or element value to add
+	 * @param list The list to add toto add to
```
Comment (ern) [inline assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentSubmissionAccess.java]: typo

File: assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java [hunk 2]
```diff
@@ -10246,6 +10246,45 @@ public void valueUnbound(SessionBindingEvent event)
 		
 		protected Assignment m_asn;
 		
+		/**
+		 * Helper method to add elements or attributse to a list
+		 * @param attributeName Name of the attribute or element value to add
+		 * @param list The list to add toto add to
```
Comment (ern) [inline assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java]: typo

**Comment by ern** [comment]
Nice refactor!

**Comment by ottenhoff** [comment]
Is this ready to go?

**Comment by jonespm** [comment]
Let me double check for Earle's review then I'll commit

**Comment by bjones86** [comment]
Pretty sure it's good, but I'd like to leave it open for a few more days (contrary to my earlier comment). We're just starting a round of internal QA, so just in case we find any issues we can leave this open (and the other related PRs).

**Pull request merged**